.github/workflows/nightly.yml

```diff
@@ -14,10 +14,20 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
-      - run: |
+      - name: Install Python requirements
+        run: |
           python -m pip install -r requirements/release.txt
-          python -m build --sdist --wheel
-      - uses: scientific-python/upload-nightly-action@main
+      - name: Build a wheel and a sdist
+        run: |
+          PYTHONWARNINGS=error,default::DeprecationWarning python -m build .
```

Comment:
Actually, I'll back off on the warning check until another PR as this currently fails with

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 287, in get_requires_for_build_sdist
    return hook(config_settings)
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/build_meta.py", line 344, in get_requires_for_build_sdist
    return self._get_build_requires(config_settings, requirements=[])
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
    self.run_setup()
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/build_meta.py", line 338, in run_setup
    exec(code, locals())
  File "<string>", line 1, in <module>
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/__init__.py", line 107, in setup
    return distutils.core.setup(**attrs)
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/_distutils/core.py", line [15](https://github.com/matthewfeickert/networkx/actions/runs/5733349767/job/15537906367#step:5:16)9, in setup
    dist.parse_config_files()
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/dist.py", line 898, in parse_config_files
    pyprojecttoml.apply_configuration(self, filename, ignore_option_errors)
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 66, in apply_configuration
    config = read_configuration(filepath, True, ignore_option_errors, dist)
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py", line 111, in read_configuration
    _BetaConfiguration.emit()
  File "/tmp/build-env-hk4hq5ip/lib/python3.10/site-packages/setuptools/warnings.py", line 47, in emit
    warnings.warn(text, cls, stacklevel=stacklevel + 1)
setuptools.config.pyprojecttoml._BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
```